### PR TITLE
handling broader set of incoming validation messages

### PR DESCRIPTION
**Pull request by slevertbiot**

### CODE REVIEW

File: udmif/event-handler/src/model/Validation.ts [hunk 1]
```diff
@@ -20,69 +20,87 @@ import { InvalidMessageError } from "../InvalidMessageError";
...
-        return this._document;
+  detail(detail: string): ValidationBuilder {
+    if (detail) {
+      this._detail = detail;
     }
+    return this;
+  }
+
+  build(): Validation {
+    this.timestamp(this._timestamp);
```
Comment (mspratt-biot) [inline udmif/event-handler/src/model/Validation.ts:93]: How come these two are done in here?

File: udmif/event-handler/src/model/Validation.ts [hunk 2]
```diff
@@ -20,69 +20,87 @@ import { InvalidMessageError } from "../InvalidMessageError";
...
+    this.version(this._version);
+
+    return {
```
Comment (mspratt-biot) [inline udmif/event-handler/src/model/Validation.ts:96]: I get the builder pattern, I think I've mentioned this before. Whenever you use this builder, you have to chain call each method to set. Wouldn't it make more sense to have a class that had a constructor which did the chaining internally here? So whenever you called `new Validation(item)`, you just get the built object immediately? Is there ever a use case for building this Validation object where we would not want to set a certain field?

Comment (slevertbiot) [reply udmif/event-handler/src/model/Validation.ts:96]: That is a good point.  In this scenario, we usually want all the fields.  I'll try reworking this to a public static type of method that returns a Validation object.

Comment (slevertbiot) [reply udmif/event-handler/src/model/Validation.ts:96]: Something like: 

```
export function fromValidationMessage(validationMessage: ValidationMessage): Validation {
    return new ValidationBuilder()
        .timestamp(validationMessage.data.timestamp)
        .version(validationMessage.data.version)
        .status(validationMessage.data.status)
        .category(validationMessage.data.status.category)
        .message(validationMessage.data.status.message)
        .detail(validationMessage.data.status.detail)
        .errors(validationMessage.data.errors)
        .build();
}
```

Comment (slevertbiot) [reply udmif/event-handler/src/model/Validation.ts:96]: So there will be cases where the Validation object will not have all the fields filled in.  I was handling that with internal logic at each method.

Comment (slevertbiot) [reply udmif/event-handler/src/model/Validation.ts:96]: That is what the DeviceDocumentFactory is doing. 😄

**Review by mspratt-biot — APPROVED** [review]

**Pull request merged**